### PR TITLE
Add hostnames of other nodes to host aliases

### PR DESCRIPTION
**Pull request by kaweue**

### CODE REVIEW

**Comment by kaweue** [comment]
recheck

**Comment by Dysproz** [comment]
recheck ha

File: deploy/1-create-operator.yaml [hunk 1]
```diff
@@ -79,7 +79,7 @@ rules:
 - apiGroups:
   - contrail
   resources:
-  - '*'
+  - contrailcnis
```
Comment (Dysproz) [inline deploy/1-create-operator.yaml:82]: Why this change?

File: pkg/apis/contrail/v1alpha1/base_types.go [hunk 2]
```diff
@@ -100,6 +100,12 @@ type PodConfiguration struct {
 	// +k8s:conversion-gen=false
 	// +optional
 	HostNetwork *bool `json:"hostNetwork,omitempty" protobuf:"varint,11,opt,name=hostNetwork"`
+	// HostAliases is an optional list of hosts and IPs that will be injected into the pod's hosts
+	// file if specified. This is only valid for non-hostNetwork pods.
```
Comment (katrybacka) [inline pkg/apis/contrail/v1alpha1/base_types.go:104]: non-hostNetwork pods?

Comment (kaweue) [reply pkg/apis/contrail/v1alpha1/base_types.go:104]: It looks like this comment is not true. It was copy passed from core/v1/types.go.
However this is supported for hostnetworking:
https://github.com/kubernetes/kubernetes/commit/4edd92f26d812c2bc7b7442e6a553ca3f60a4a8e

Comment (kaweue) [reply pkg/apis/contrail/v1alpha1/base_types.go:104]: I will delete that sentence from this comment

File: deploy/1-create-operator.yaml (hunk 1 above)
Comment (kaweue) [reply deploy/1-create-operator.yaml:82]: I just run ./create_manifest.sh

Comment (kaweue) [reply deploy/1-create-operator.yaml:82]: I will change that in deploy/role.yaml

**Comment by kaweue** [comment]
recheck ha

File: pkg/apis/contrail/v1alpha1/base_types.go (hunk 2 above)
Comment (kaweue) [reply pkg/apis/contrail/v1alpha1/base_types.go]: done

File: deploy/1-create-operator.yaml (hunk 1 above)
Comment (kaweue) [reply deploy/1-create-operator.yaml]: done

**Review by katrybacka — APPROVED** [review]

**Comment by psykulsk** [comment]
recheck ha

**Review by Dysproz — APPROVED** [review]
